src/System.CommandLine/Binding/ModelBinder{T}.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq.Expressions;
+
+namespace System.CommandLine.Binding
+{
+    public class ModelBinder<TModel> : ModelBinder
+    {
+        public ModelBinder() : base(ModelDescriptor.FromType<TModel>())
+        {
+        }
+
+        public void BindMemberFromOption<TValue>(
+            Expression<Func<TModel, TValue>> property,
+            IOption option)
+        {
+            NamedValueSources.Add(
+                property.MemberTypeAndName(),
+                new SpecificSymbolValueSource(option));
+        }
+
+        public void BindMemberFromCommand<TValue>(
```

Comment:
Does this mean BindMemberFromArgument? How will a tight coupling between commands and arguments here work with multiple arguments? I do not believe people think about commands having values as much as having arguments.

Reply:
Ignore the above because this isn't the place for it, but to clarify the confusion. 

Our implementation is at odds with the ways I believe people think about the problem, and reflects an implementation detail. I do not believe people say "What's the value of that command?" Nor "What's the value of the argument on that option?" I think they would say "What's the value of the argument?" or "What's the value of the option?" and that is how Help seems laid out to me. I think this is going to be a rough cognitive load and reflects an implementation detail, and I know I will keep making this mistake whenever I am thinking about the real problem (the CLI). 

But this has nothing to do with the problem we are trying to solve today.

Reply:
It does mean bind from the argument. The concern is a good one. I think conceptually "What is the value for this option" is more intuitive but less precise than "What is the value for this option's argument"? And yes, the value for a command will become more ambiguous with #310.